sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -362,7 +362,7 @@ object DottyPlugin extends AutoPlugin {
 
       // Configuration for the doctool
       resolvers ++= (if(!useScala3doc.value) Nil else Seq(Resolver.jcenterRepo)),
-      useScala3doc := false,
+      useScala3doc := VersionNumber(scalaVersion.value).matchesSemVer(SemanticSelector(">=3.0.0-M3")),
```

Comment:
Note that I haven't checked if the current version number is considered `>= 3.0.0-M3`